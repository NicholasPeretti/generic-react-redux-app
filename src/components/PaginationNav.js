import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Text from './Text'
import Style from '../style'
import _StyledLink from './StyledLink'

const LABELS_SPACING = Style.spacing.padding2

const NavContainer = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: ${Style.spacing.padding3} 0px;
`

const StyledLink = styled(_StyledLink)`
  margin: 0px ${LABELS_SPACING};
`

var PaginationNav = ({ page, hasNext, hasPrev, totalPages, fetchPage }) => (
  <NavContainer>
    <Text>
      <StyledLink onClick={() => fetchPage(1)}>First</StyledLink>

      <StyledLink
        onClick={() => {
          if (hasPrev) fetchPage(page - 1)
        }}
      >
        Previous
      </StyledLink>

      <span style={{ margin: `0px ${LABELS_SPACING}` }}>
        Page {page} of {totalPages}
      </span>

      <StyledLink
        onClick={() => {
          if (hasNext) fetchPage(page + 1)
        }}
      >
        Next
      </StyledLink>

      <StyledLink onClick={() => fetchPage(totalPages)}>Last</StyledLink>
    </Text>
  </NavContainer>
)

PaginationNav.propTypes = {
  page: PropTypes.number.isRequired,
  hasNext: PropTypes.bool.isRequired,
  hasPrev: PropTypes.bool.isRequired,
  totalPages: PropTypes.number.isRequired,
  fetchPage: PropTypes.func.isRequired
}

export default PaginationNav
