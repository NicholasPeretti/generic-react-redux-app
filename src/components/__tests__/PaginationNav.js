import React from 'react'
import { shallow } from 'enzyme'
import PresentationalTest from '../../utils/tests/presentational'
import PaginationNav from '../PaginationNav'

const props = {
  page: 1,
  hasNext: false,
  hasPrev: false,
  totalPages: 1,
  fetchPage: jest.fn()
}

describe('PaginationNav', () => {
  PresentationalTest(PaginationNav, props)

  test('Should match snapshot', () => {
    const wrapper = shallow(<PaginationNav {...props}/>)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
