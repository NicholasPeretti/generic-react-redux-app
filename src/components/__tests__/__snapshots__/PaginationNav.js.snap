// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNav Should match snapshot 1`] = `"<nav class=\\"sc-htpNat cZatAP\\"><span class=\\"sc-bdVaJa bkmfbd\\"><a class=\\"sc-bwzfXH sc-bxivhb hyEKhL\\">First</a><a class=\\"sc-bwzfXH sc-bxivhb hyEKhL\\">Previous</a><span style=\\"margin:0px 12px\\">Page 1 of 1</span><a class=\\"sc-bwzfXH sc-bxivhb hyEKhL\\">Next</a><a class=\\"sc-bwzfXH sc-bxivhb hyEKhL\\">Last</a></span></nav>"`;
